jdk-http-client/src/main/scala/org/http4s/client/jdkhttpclient/JdkHttpClient.scala

```diff
@@ -0,0 +1,111 @@
+package org.http4s.client.jdkhttpclient
+
+import java.net.URI
+import java.net.http.HttpRequest.BodyPublishers
+import java.net.http.HttpResponse.BodyHandlers
+import java.net.http.{HttpClient, HttpRequest, HttpResponse}
+import java.nio.ByteBuffer
+import java.util
+import java.util.concurrent.{CancellationException, CompletableFuture, CompletionException, Flow}
+import java.util.function.BiFunction
+
+import cats.effect._
+import cats.implicits._
+import fs2.interop.reactivestreams._
+import fs2.{Chunk, Stream}
+import org.http4s.client.Client
+import org.http4s.util.CaseInsensitiveString
+import org.http4s.{Header, Headers, HttpVersion, Request, Response, Status}
+import org.reactivestreams.FlowAdapters
+
+import scala.collection.JavaConverters._
+
+object JdkHttpClient {
+
+  def apply[F[_]](builder: HttpClient.Builder = HttpClient.newBuilder)(
+      implicit F: ConcurrentEffect[F]): F[Client[F]] =
+    F.delay(builder.build).map { jdkHttpClient =>
+      Client[F] { req =>
+        Resource.liftF {
+          F.delay(jdkHttpClient.sendAsync(convertRequest(req), BodyHandlers.ofPublisher))
+            .flatMap(fromJavaFuture(_))
+            .flatMap(convertResponse(_))
+        }
+      }
+    }
+
+  private def fromJavaFuture[F[_], A](cf: CompletableFuture[A])(implicit F: Concurrent[F]): F[A] =
```

Comment:
This might be good in org.http4s.internal in core.  I can imagine other backends benefiting from this.

Reply:
This is a slight adaption of [this](https://github.com/typelevel/cats-effect/issues/160), and it doesn't seem planned to add it to cats-effect in the near future, so it makes sense to make it an http4s internal utility.